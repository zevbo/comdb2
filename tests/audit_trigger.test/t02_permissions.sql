CREATE TABLE t(i int)$$
CREATE LUA AUDIT TRIGGER bloop ON (TABLE t FOR INSERT)
SELECT "-----------permission denied----------"
ALTER TABLE "$audit_t" ADD COLUMN b int$$
ALTER TABLE "$audit_t" { schema { int b } }$$
DROP TABLE "$audit_t"$$
ALTER TABLE "$audit_t" RENAME TO blooper$$
ALTER TABLE "$audit_t" DROP COLUMN old_i$$
ALTER TABLE "$audit_t" ADD UNIQUE INDEX "hello" (new_i, old_i)$$

DROP LUA TRIGGER bloop

SELECT sleep(1)
SELECT "-----------permission approved----------"
ALTER TABLE "$audit_t" ADD COLUMN b int$$
ALTER TABLE "$audit_t" RENAME TO blooper$$
ALTER TABLE blooper DROP COLUMN old_i$$
ALTER TABLE blooper ADD UNIQUE INDEX "hello" (new_i, old_i)$$
ALTER TABLE blooper { schema { int b } }$$

DROP TABLE t 
DROP TABLE blooper

