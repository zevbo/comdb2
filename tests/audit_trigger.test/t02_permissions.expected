("-----------permission denied----------"='-----------permission denied----------')
[ALTER TABLE "$audit_t" ADD COLUMN b int] failed with rc -3 permission denied to alter schema of $audit_t
[ALTER TABLE "$audit_t" { schema { int b } }] failed with rc -3 permission denied to alter schema of $audit_t
[DROP TABLE "$audit_t"$$] failed with rc -3 near "$$": syntax error
[ALTER TABLE "$audit_t" RENAME TO blooper] failed with rc -3 permission denied to alter name of $audit_t
[ALTER TABLE "$audit_t" DROP COLUMN old_i] failed with rc -3 permission denied to alter schema of $audit_t
[ALTER TABLE "$audit_t" ADD UNIQUE INDEX "hello" (new_i, old_i)] failed with rc -3 permission denied to alter schema of $audit_t
("-----------permission approved----------"='-----------permission approved----------')
[ALTER TABLE "$audit_t" RENAME TO blooper] failed with rc 240 a schema change error occurred
[ALTER TABLE blooper DROP COLUMN old_i] failed with rc -3 Table not found
[ALTER TABLE blooper ADD UNIQUE INDEX "hello" (new_i, old_i)] failed with rc -3 Table not found
[ALTER TABLE blooper { schema { int b } }] failed with rc -3 Table not found
[DROP TABLE blooper] failed with rc -3 no such table: blooper
